proposed/simplecache.md

```diff
@@ -0,0 +1,230 @@
+libraries. Interoperability at this level means libraries can drop their
+own caching implementations and easily rely on the one given to them by the
+framework, or another dedicated cache library.
+
+PSR-6 solves this problem already, but in a rather formal and verbose way for
+what the most simple use cases need. This simpler approach aims to build a
+standardized layer of simplicity on top of the existing PSR-6 interfaces.
+
+
+### 1.2 Definitions
+
+Definitions for Calling Library, Implementing Library, TTL, Expiration and Key
+are copied from PSR-6 as the same assumptions are true.
+
+*    **Calling Library** - The library or code that actually needs the cache
+services. This library will utilize caching services that implement this
+standard's interfaces, but will otherwise have no knowledge of the
+implementation of those caching services.
+
+*    **Implementing Library** - This library is responsible for implementing
+this standard in order to provide caching services to any Calling Library. The
+Implementing Library MUST provide a class implementing the Psr\SimpleCache\CacheInterface interface.
+Implementing Libraries MUST support at minimum TTL functionality as described
+below with whole-second granularity.
+
+*    **TTL** - The Time To Live (TTL) of an item is the amount of time between
+when that item is stored and it is considered stale. The TTL is normally defined
+by an integer representing time in seconds, or a DateInterval object.
+
+*    **Expiration** - The actual time when an item is set to go stale. This it
+typically calculated by adding the TTL to the time when an object is stored, but
+may also be explicitly set with DateTime object.
+
+    An item with a 300 second TTL stored at 1:30:00 will have an expiration of
+    1:35:00.
+
+    Implementing Libraries MAY expire an item before its requested Expiration Time,
+but MUST treat an item as expired once its Expiration Time is reached. If a calling
+library asks for an item to be saved but does not specify an expiration time, or
+specifies a null expiration time or TTL, an Implementing Library MAY use a configured
+default duration. If no default duration has been set, the Implementing Library
+MUST interpret that as a request to cache the item forever, or for as long as the
+underlying implementation supports.
+
+*    **Key** - A string of at least one character that uniquely identifies a
+cached item. Implementing libraries MUST support keys consisting of the
+characters `A-Z`, `a-z`, `0-9`, `_`, and `.` in any order in UTF-8 encoding and a
+length of up to 64 characters. Implementing libraries MAY support additional
+characters and encodings or longer lengths, but must support at least that
+minimum.  Libraries are responsible for their own escaping of key strings
+as appropriate, but MUST be able to return the original unmodified key string.
+The following characters are reserved for future extensions and MUST NOT be
```

Comment:
It's common to `json_encode` or `md5` structures to get their keys. That would prevent such use. Is it intended?